### ISSUE TITLE
ImageEditor is very useful, but upload a image takes 5 second and other issue.

### ISSUE THREAD

**Issue by Dexter-Wang** [gradio-app/gradio#6740]
### Describe the bug

gr.ImageEditor is very useful, but upload a image takes 5 second sometimes.

There is another issue as bellow.
Transfer Image from the gr.Image at one Tab to gr.ImageEditor at another Tab, it got black image quite often.    

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

import torch
from diffusers import StableDiffusionImg2ImgPipeline   
from diffusers import DiffusionPipeline         

NegPrompt ="""low res, bad hands, text, error, missing fingers, extra digit, fewer digits, cropped, 
              worst quality, low quality, normal quality, jpeg artifacts, signature, watermark,
              username, blurry, ugly""" 

modelID1 = "pagebrain/dreamshaper-v8"                
Img2Img1 = StableDiffusionImg2ImgPipeline.from_pretrained(modelID1, torch_dtype=torch.float16,
                                          safety_checker=None, requires_safety_checker=False).to("cuda")

'''--- ImageToImage  --------------------------------------------- '''   
def ImageToImage0(imgIn=None, prompt="", strength=0.75, seed=0): 
    if (imgIn==None or prompt==""): return None              
    gene = torch.Generator(device="cuda").manual_seed(0) 
    imgOut = Img2Img1(prompt=prompt, negative_prompt=NegPrompt, image=imgIn, 
                          generator=gene, strength=strength, guidance_scale=7.5).images[0]  
    return imgOut  

'''--- TextToImage  --------------------------------------------- '''
Text2Img1 = DiffusionPipeline.from_pretrained(modelID1, torch_dtype=torch.float16,
              safety_checker=None, requires_safety_checker=False).to("cuda") 

def TextToImage0(prompt=""): 
    if (prompt==""): return None
    image = Text2Img1(prompt, negative_prompt=NegPrompt).images[0]         
    return image  

'''--- Create UI --------------------------------------------- '''
import numpy as np
from PIL import ImageFilter
import gradio as gr

def ImgToInput(imgOut, imgIn):   
    if (imgOut==None): return None
    imgIn["background"]=imgOut.copy()  
    return imgIn    

def ImageToImage(imgIn, prompt, strength):  
    if (imgIn==None): return None 
    imgIn = imgIn["composite"].copy()           
    imgOut  = ImageToImage0(imgIn, prompt, strength)    
    return imgOut 

def TextToImage(prompt):
    if (prompt==None): return None
    imgOut  = TextToImage0(prompt)     
    return imgOut 

def ImageBlur(imgIn, radius=30):    
    imgIn  = imgIn["composite"].copy()  
    imgOut = imgIn.filter(ImageFilter.GaussianBlur(radius=radius))   
    return imgOut

Text1  = "A beautiful country side landscape, romantic"
with gr.Blocks() as MainForm:   
    Prompt = gr.Textbox(value=Text1, label='Prompt')           
    with gr.Tab("ImageToImage"):
        with gr.Row():                               
            with gr.Column():                         
                IgClear = gr.Button("Clear")                   
                IgImgIn = gr.ImageEditor(type='pil', image_mode="RGB", width=512, height=512, interactive=True, )  
                Strength  = gr.Slider(0.1, 1.0, 0.75, step=0.01, label='Strength')                                   
            with gr.Column():                           
                Ig2ImgGN  = gr.Button("Generate") 
                IgImgOut  = gr.Image(type='pil', width=512, height=512, )    
                IgToInput = gr.Button("To Input")  

    with gr.Tab("TextToImage"):      
        with gr.Row():                               
            with gr.Column():                         
                TxClear = gr.Button("Clear")                                  
            with gr.Column():                         
                Tx2ImgGN = gr.Button("Generate") 
                TxImgOut = gr.Image(type='pil', width=512, height=512, )
                TxToIgIn = gr.Button("To ImageToImage Input") 
                
    with gr.Tab("Preprocess"):      
        with gr.Row():                              
            with gr.Column():                        
                PrClear = gr.Button("Clear") 
                PrImgIn = gr.ImageEditor(type='pil', image_mode="RGB",  width=512, height=512, interactive=True, )                        
                with gr.Row():                      
                    BlurNm  = gr.Number(minimum=0, maximum=50, value=30, step=5, label="Blur", interactive=True, show_label=False)                     
                    BlurBtn = gr.Button("Blur") 
            with gr.Column():                                                        
                PrImgOut  = gr.Image(type='pil', width=512, height=512, )
                with gr.Row():
                    PrToInput = gr.Button("To Input") 
                    PrToIgIn  = gr.Button("To IamgeToImage Input") 
       
    #---  ImgToImg ------ 
    IgClear.click(fn=lambda:[None,None], inputs=[], outputs=[IgImgIn,IgImgOut])           
    Ig2ImgGN.click(fn=ImageToImage, inputs=[IgImgIn,Prompt,Strength], outputs=IgImgOut) 
    IgToInput.click(fn=ImgToInput, inputs=IgImgOut, outputs=IgImgIn) 
    
    #---  TextToImg ------    
    TxClear.click(fn=lambda:[None,None], inputs=[], outputs=[TxImgOut])        
    Tx2ImgGN.click(fn=TextToImage, inputs=Prompt, outputs=TxImgOut)
    TxToIgIn.click(fn=ImgToInput, inputs=[TxImgOut,IgImgIn], outputs=IgImgIn) 
    
    #---  Preprocess ------    
    BlurBtn.click  (fn=ImageBlur, inputs=[PrImgIn,BlurNm], outputs=PrImgOut)
    PrToInput.click(fn=ImgToInput, inputs=[PrImgOut,PrImgIn], outputs=PrImgIn)      
    PrToIgIn.click (fn=ImgToInput, inputs=[PrImgOut,IgImgIn], outputs=IgImgIn)   
MainForm.launch()   

### Screenshot

![ImageEditor_issue](https://github.com/gradio-app/gradio/assets/64398365/844a56a8-68c0-4d6d-b666-4e16382b0faf)

### Logs

```shell
no error
```

### System Info

```shell
gradio                       4.7.1
gradio_client                0.7.0
python-dateutil              2.8.2
python-multipart             0.0.6
```

### Severity

I can work around it

**Comment by pngwn** [comment]
Do you have a minimal reproduction demonstrating this issue? There is a lot of unrelated code in the snippet you posted.

**Comment by pngwn** [comment]
cc @aliabid94 re: perf improvements.

**Comment by Dexter-Wang** [comment]
import torch
from diffusers import StableDiffusionImg2ImgPipeline  

modelID1 = "pagebrain/dreamshaper-v8"                
Img2Img1 = StableDiffusionImg2ImgPipeline.from_pretrained(modelID1, torch_dtype=torch.float16,
                                          safety_checker=None, requires_safety_checker=False).to("cuda")

'''--- ImageToImage  --------------------------------------------- '''   
def ImageToImage0(imgIn=None, prompt="", strength=0.75, seed=0): 
    if (imgIn==None or prompt==""): return None              
    gene = torch.Generator(device="cuda").manual_seed(0) 
    imgOut = Img2Img1(prompt=prompt, image=imgIn, generator=gene, strength=strength, guidance_scale=7.5).images[0]  
    return imgOut  

'''--- Create UI --------------------------------------------- '''
from PIL import ImageFilter
import gradio as gr

def ImgToInput(imgOut, imgIn):   
    if (imgOut==None): return None
    imgIn["background"]=imgOut.copy()  
    return imgIn    

def ImageToImage(imgIn, prompt):  
    if (imgIn==None): return None 
    imgIn = imgIn["composite"].copy()           
    imgOut  = ImageToImage0(imgIn, prompt)    
    return imgOut 

def ImageBlur(imgIn, radius=30):    
    imgIn  = imgIn["composite"].copy()  
    imgOut = imgIn.filter(ImageFilter.GaussianBlur(radius=radius))   
    return imgOut

with gr.Blocks() as MainForm:   
    Prompt = gr.Textbox(value="A beautiful country side landscape, romantic", label='Prompt')           
    with gr.Tab("ImageToImage"):
        with gr.Row():                               
            with gr.Column():                         
                IgClear = gr.Button("Clear")                   
                IgImgIn = gr.ImageEditor(type='pil', image_mode="RGB", width=512, height=512, interactive=True, ) 
            with gr.Column():                           
                Ig2ImgGN  = gr.Button("Generate") 
                IgImgOut  = gr.Image(type='pil', width=512, height=512, ) 
                 
    with gr.Tab("Preprocess"):      
        with gr.Row():                              
            with gr.Column():                        
                PrClear = gr.Button("Clear") 
                PrImgIn = gr.ImageEditor(type='pil', image_mode="RGB",  width=512, height=512, interactive=True, ) 
                BlurBtn = gr.Button("Blur") 
            with gr.Column():                                                        
                PrImgOut  = gr.Image(type='pil', width=512, height=512, )
                with gr.Row():                    
                    PrToIgIn  = gr.Button("To IamgeToImage Input") 
       
    #---  ImgToImg ------ 
    IgClear.click(fn=lambda:[None,None], inputs=[], outputs=[IgImgIn,IgImgOut])           
    Ig2ImgGN.click(fn=ImageToImage, inputs=[IgImgIn,Prompt], outputs=IgImgOut) 
    
    #---  Preprocess ------   
    PrClear.click(fn=lambda:[None,None], inputs=[], outputs=[PrImgIn,PrImgOut])
    BlurBtn.click  (fn=ImageBlur, inputs=PrImgIn, outputs=PrImgOut)    
    PrToIgIn.click (fn=ImgToInput, inputs=[PrImgOut,IgImgIn], outputs=IgImgIn)   
MainForm.launch()

**Comment by Dexter-Wang** [comment]
Thank you for your quick reply.  Above is a simplify code.  Black image happen less.  Following is the working flow to make Black image happen.

1. upload a image to the input of "Preprocess", press "Blur", then a blur image generated at "Output". It work fine.
2. Press "To ImageToImage Input".  It work fine.
3. At "Image To Image" page, input prompt and press "Generate",  It work fine.   
4. At  "Image To Image" page,  upload a image to the input frame, press "Generate".   Problem happen at this moment.  The input  frame either shows the wrong image or the black image.

however, ImageEditor is a very good design.

**Comment by Dexter-Wang** [comment]
I also put the above simplify code to the Colab.  Following is the address if you need. 
https://drive.google.com/drive/my-drive

**Comment by Dexter-Wang** [comment]
I also put the above code at Colab, Following is the address, if you need it.

https://colab.research.google.com/drive/1QhhGt955SkCrlGc8TJyLYDZkiJ2-EGDr

**Comment by abidlabs** [comment]
Hi, apologies for the late follow up. We haven't had a chance to look into this issue, but the Gradio codebase has changed quite significantly since this issue was created, particularly with the release of Gradio 5. Could you let us know if this is still an issue in the latest version of Gradio (`pip install --upgrade gradio`)? Thanks!

**Comment by Dexter-Wang** [comment]
Many thanks for reply, I shall try in this few days.從我的iPhone傳送Abubakar Abid ***@***.***> 於 2024年10月18日 上午5:30 寫道：﻿
Hi, apologies for the late follow up. We haven't had a chance to look into this issue, but the Gradio codebase has changed quite significantly since this issue was created, particularly with the release of Gradio 5. Could you let us know if this is still an issue in the latest version of Gradio (pip install --upgrade gradio)? Thanks!

**Comment by Dexter-Wang** [comment]
I had put the above code at Colab, Following is the address, if you need it.

https://colab.research.google.com/drive/1QhhGt955SkCrlGc8TJyLYDZkiJ2-EGDr

I try again with gradio-5.1.0 by same code at above Colab, following is the issue.  

  1. upload an image to the input of "Preprocess" page, press "Blur", then a blur image generated at "Output"(right hand side). It worked fine.
  2. Press the button of "To ImageToImage Input"(at the bottom of right hand side).  It got a black screen at the input of ImageToImage page.  The input is at left hand side.

![1_Preprocess](https://github.com/user-attachments/assets/a811b32e-4599-4412-94c9-8736aa815f04)
![2_ImageToImage](https://github.com/user-attachments/assets/c47221e2-0745-49f7-9fe6-aca26ec6b227)
